core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -346,6 +348,112 @@ public void awaitTxn(String tableName, long txn, long timeout, TimeUnit timeoutU
+                            reader.of(path.$());
+                            MatViewRefreshState.readFrom(reader, state);
+                        }
+
+                        if (state != null && !state.isInvalid()) {
+                            MatViewDefinition matViewDefinition = state.getViewDefinition();
+                            final TableToken baseTableToken = tableNameRegistry.getTableToken(matViewDefinition.getBaseTableName());
+                            final boolean baseTableExists = baseTableToken != null && !tableNameRegistry.isTableDropped(baseTableToken);
+                            if (!baseTableExists) {
+                                // Print a warning, but let the mat view load in invalid state.
+                                LOG.info().$("base table for materialized view does not exist [table=").utf8(matViewDefinition.getBaseTableName())
+                                        .$(", view=").utf8(tableToken.getTableName())
+                                        .I$();
+                            }
+                            // this should be called only during replica to primary transition
+                            if (checkMatViewConsistency) {
+
+                                long matViewLastTxn = getTableSequencerAPI().lastTxn(tableToken);
+                                long baseTableLastTxn = baseTableExists && baseTableToken.isWal()
+                                        ? getTableSequencerAPI().lastTxn(baseTableToken)
+                                        : Long.MAX_VALUE;
+                                if (state.getSeqTxn() > matViewLastTxn || state.getLastRefreshBaseTxn() > baseTableLastTxn) {
+                                    // Materialized view refresh state is ahead of the base table or the view itself.
+                                    // This may happen when a read-only replica node which isn't fully caught up
+                                    // was converted to primary.
+                                    try (TransactionLogCursor transactionLogCursor = getTableSequencerAPI().getCursor(tableToken, matViewLastTxn - 1)) {
+                                        // slow-path, search for the last refresh base txn in the wal-e file
+                                        long refreshBaseTxn = -1;
+                                        try {
+                                            refreshBaseTxn = WalUtils.getRefreshTxn(
+                                                    path.trimTo(pathLen).concat(tableToken.getDirName()),
+                                                    transactionLogCursor,
+                                                    configuration.getFilesFacade()
+                                            );
+                                        } catch (CairoException e) {
+                                            LOG.error().$("could not find last refresh base txn for materialized view [view=")
```

Comment:
What happens if this is the case? Can you log what are the consequences of this? Why there is no mat view invalidation anywhere in this consistency check?